base/ca/src/org/dogtagpki/ct/CTEngine.java

```diff
@@ -0,0 +1,745 @@
+                    boolean allowFailedSCTVerification = true;
+
+                    boolean verified = verifySCT(CTResponse.fromJSON(respS), tbsCert, ls.getPublicKey(), ctCA);
+                    if (verified) {
+                        logger.info(method + "verifySCT returned true; SCT is valid");
+                    } else {
+                        // log at WARN if !verified, regardless of how we are treating
+                        // failed verifications, because it is indicative of log server
+                        // misbehavoiur
+                        logger.warn(method + "verifySCT returns false; SCT failed to verify");
+                    }
+                    if (verified || allowFailedSCTVerification) {
+                        ctResponses.add(respS);
+                    } else {
+                        throw new EBaseException(errMsg);
+                    }
+                }
+
+                /**
+                 * Now onto turning the precert into a real cert with the SCT list extension
+                 */
+
+                // create SCT extension
+                Extension sctExt = createSCTextension(ctResponses);
+                if (sctExt == null) {
+                    errMsg = " createSCTextension returns null";
+                    logger.debug(method + errMsg);
+                    throw new EBaseException(errMsg);
+                }
+
+                // add the SCT extension
+                exts.set(sctExt.getExtensionId().toString(), sctExt);
+
+                certi.delete(X509CertInfo.EXTENSIONS);
+                certi.set(X509CertInfo.EXTENSIONS, exts);
+                printExtensions(exts);
+            }
+        } catch (Exception e) {
+            logger.error(method + "Error occurred: " + e.getMessage(), e);
+            throw new EBaseException(e.getMessage());
+        }
+    }
+
+    /*
+     * for debugging
+     */
+    protected void printExtensions(CertificateExtensions exts) {
+
+        String method = "CAService.printExtensions: ";
+        logger.debug(method + "begins");
+        try {
+            if (exts == null)
+                return;
+
+            Enumeration<String> e = exts.getNames();
+            while (e.hasMoreElements()) {
+                String n = e.nextElement();
+                Extension ext = (Extension) exts.get(n);
+
+                logger.debug(" ---- " + ext.getExtensionId().toString());
+            }
+        } catch (Exception e) {
+            logger.debug(method + e.toString());
+        }
+        logger.debug(method + "ends");
+    }
+
+    /**
+     * (Certificate Transparency)
+     *
+     * timeStampHexStringToByteArray converts timestamp hex string to bytes
+     *   - example timestamp: 00000172.71270909
+     * @param timeStampString hex string (example above)
+     * @return byte[] timestamp in byte array
+     */
+    public static byte[] timeStampHexStringToByteArray(String timeStampString) {
```

Comment:
Minor: Needs to be moved to a different class, as these are CT specific util methods.

Reply:
sounds good!